File: Iteration2/src/Domain/CourseSection.java
package Domain;
import java.util.ArrayList;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class CourseSection{

    private String courseSectionNo;
    private int noOfStudents;
    private int Quota;
    private ArrayList<String> sectionTime;
    private String courseCode;


    public CourseSection(String courseCode, String courseSectionNo, int Quota, ArrayList<String> sectionTimes){
        this.noOfStudents = 0;
        this.courseCode = courseCode;
        this.courseSectionNo = courseSectionNo;
        this.Quota = Quota;
        this.sectionTime = sectionTimes;
    }
    public CourseSection(){};


    public void addStudent(){
        noOfStudents++;
    }

    public void printSections(){
        System.out.println("Course Section No: " + getCourseSectionNo() + " Quota: " + getQuota() + " No of Students: " + getNoOfStudents() + " Lecture Times:");
        for(String each : sectionTime){
            System.out.println(getDay(each)+ " " + each.substring(1,3) + ":00");
        }
        System.out.println();
    }

    @JsonIgnore
    public boolean isFull(){
        return noOfStudents == Quota;
    }

    public String getcourseCode() {
        return courseCode;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public String getCourseSectionNo() {
        return courseSectionNo;
    }

    public void setCourseSectionNo(String courseSectionNo) {
        this.courseSectionNo = courseSectionNo;
    }

    public int getNoOfStudents() {
        return noOfStudents;
    }
    
    public void setNoOfStudents(int noOfStudents) {
        this.noOfStudents = noOfStudents;
    }

    public int getQuota() {
        return Quota;
    }

    public void setQuota(int Quota) {
        this.Quota = Quota;
    }

    public ArrayList<String> getSectionTime() {
        return sectionTime;
    }

    public void setSectionTime(ArrayList<String> sectionTimes) {
        this.sectionTime = sectionTimes;
    }

    private String getDay(String time){
        String dayNo = time.substring(0,1);
        switch(dayNo){
            case "1":
                return "Monday";
            case "2":
                return "Tuesday";

            case "3":
                return "Wednesday";

            case "4":
                return "Thursday";
            
            case "5":
                return "Friday";
            
            default:
                return "Free Day";
        }
    }

}

File: Iteration3/Domain/CourseSection.py
from typing import List

class CourseSection:
    def __init__(self, course_code, course_section_no, quota, section_times):
        self.no_of_students = 0
        self.course_code = course_code
        self.course_section_no = course_section_no
        self.quota = quota
        self.section_times = section_times


    def add_student(self):
        self.no_of_students += 1

    def print_sections(self):
        print(f"Course Section No: {self.course_section_no} Quota: {self.quota} No of Students: {self.no_of_students} Lecture Times:")
        for time in self.section_times:
            print(f"{self.get_day(time)} {time[1:3]}:00")
        print()

    def is_full(self):
        return self.no_of_students == self.quota

    #region Properties

    def get_course_code(self):
        return self.course_code

    def set_course_code(self, course_code):
        self.course_code = course_code

    def get_course_section_no(self):
        return self.course_section_no

    def set_course_section_no(self, course_section_no):
        self.course_section_no = course_section_no

    def get_no_of_students(self):
        return self.no_of_students

    def set_no_of_students(self, no_of_students):
        self.no_of_students = no_of_students

    def get_quota(self):
        return self.quota

    def set_quota(self, quota):
        self.quota = quota

    def get_section_times(self):
        return self.section_times

    def set_section_times(self, section_times):
        self.section_times = section_times
    #endregion Properties

    def get_day(self, time):
        day_no = time[0]
        days = {
            "1": "Monday",
            "2": "Tuesday",
            "3": "Wednesday",
            "4": "Thursday",
            "5": "Friday"
        }
        return days.get(day_no, "Free Day")


import json
from Domain.CourseSection import CourseSection

class CourseSectionEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, CourseSection):
            return {
                'course_code': obj.get_course_code(),
                'course_section_no': obj.get_course_section_no(),
                'quota': obj.get_quota(),
                'section_times': obj.get_section_times()
            }
        return super().default(obj)
